src/index.test.ts

```diff
@@ -183,7 +196,10 @@ test('Should include context fields on request', async () => {
     const url = new URL(fetchMock.mock.calls[0][0]);
 
     expect(url.searchParams.get('userId')).toEqual('123');
-    expect(url.searchParams.get('randomField')).toEqual('random');
+    expect(url.searchParams.get('sessionId')).toEqual('456');
+    expect(url.searchParams.get('remoteAddress')).toEqual('address');
+    expect(url.searchParams.get('properties.property1')).toEqual('property1');
+    expect(url.searchParams.get('properties.property2')).toEqual('property2');
```

Comment:
Would be great to have a use case where properties = {} (an empty object)

Reply:
Is there a case for properties being an empty object? Would the properties field be set at all if they were empty?

Reply:
A user of the client can do whatever they want, so I would expect an empty properties object to avoid adding query params.